build-binutils.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019 The ClangBuiltLinux Authors
+# Description: Builds a standalone copy of binutils
+
+import argparse
+import multiprocessing
+import os
+import pathlib
+import platform
+import shutil
+import subprocess
+import utils
+
+
+# Convert the host architecture to the first part of a target triple
+def host_to_key():
+    if platform.machine() == "armv7l":
+        return "arm"
+    elif platform.machine() == "ppc64le":
+        return "powerpc64le"
+    elif platform.machine() == "ppc":
+        return "powerpc"
+    else:
+        return platform.machine()
+
+
+# Get the architecture from a target triple
+def target_to_arch(target):
+    return target.split("-")[0]
+
+
+# Is the current target triple the same as the host?
+def host_is_target(target):
+    return host_to_key() == target_to_arch(target)
+
+
+def parse_parameters(root):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-I",
+                        "--install-folder",
+                        help="""
+                        By default, the script will create a "usr" folder in the same folder as this script
+                        and install binutils there. If you'd like to have it installed somewhere else, pass
+                        it to this parameter. This can either be an absolute or relative path.
+                        """,
+                        type=str,
+                        default=os.path.join(root.as_posix(), "usr"))
+    parser.add_argument("-t",
+                        "--targets",
+                        help="""
+                        The script can build binutils targeting arm-linux-gnueabi, aarch64-linux-gnu,
+                        powerpc-linux-gnu, powerpc64le-linux-gnu, and x86_64-linux-gnu.
+
+                        You can either pass the full target or just the first part (arm, aarch64, x86_64, etc)
+                        or all if you want to build all targets (which is the default). It will only add the
+                        target prefix if it is not for the host architecture.
+                        """,
+                        default="all",
+                        nargs="+")
+    return parser.parse_args()
+
+
+def create_targets(targets):
+    targets_dict = {
+        "arm": "arm-linux-gnueabi",
+        "aarch64": "aarch64-linux-gnu",
+        "powerpc64le": "powerpc64le-linux-gnu",
+        "powerpc": "powerpc-linux-gnu",
+        "x86_64": "x86_64-linux-gnu"
+    }
+    targets_list = []
+
+    if ''.join(targets) == "all":
+        for key in targets_dict:
+            targets_list.append(targets_dict[key])
```

Comment:
remove the for loop, what you want here is `targets_list = list(targets_dict.keys())`

Reply:
https://github.com/MSF-Jarvis/tc-build/commit/f021dd0cb221